### ISSUE TITLE
Initial device setup hangs at 'Connecting to device...'

### ISSUE THREAD

**Issue by friedc** [tuhiproject/tuhi#229]
I have a Bamboo Spark running on Ubuntu 18.04 LTS. 'tuhi' is run via flatpak
as described in `README.md`. My bluetooth controller is an Intel 3160.

The initial dialog hangs at 'Connecting to device...'
after the blue button on the device is pressed.

`--verbose` gives me the following output in the console:

```
17:32:27 INFO: tuhi: F1:42:11:D0:3F:8B: device without config, must be registered first
17:32:28 DEBUG: tuhi: /org/bluez/hci0/dev_F1_42_11_D0_3F_8B: call Register() on device
17:32:28 DEBUG: tuhi.ble.F1:42:11:D0:3F:8B: Connecting
17:32:28 DEBUG: tuhi: /org/bluez/hci0/dev_F1_42_11_D0_3F_8B: call Register() on device
17:32:28 DEBUG: tuhi.ble.F1:42:11:D0:3F:8B: Connection established
17:32:28 ERROR: tuhi.ble.F1:42:11:D0:3F:8B: Connection failed: g-io-error-quark: GDBus.Error:org.bluez.Error.Failed: Software caused connection abort (36)
17:32:28 DEBUG: tuhi.ble.F1:42:11:D0:3F:8B: Disconnected
17:32:28 DEBUG: tuhi: F1:42:11:D0:3F:8B: disconnected
```

**Comment by whot** [comment]
Have a look at the [log files wiki page](https://github.com/tuhiproject/tuhi/wiki/Log-files) please, without the log files it'll be impossible to guess what's going on here.

**Comment by friedc** [comment]
[tuhi-19-11-11-17:32:20.log](https://github.com/tuhiproject/tuhi/files/3835142/tuhi-19-11-11-17.32.20.log)
[log-1573489775-2019-11-11-17:29:35.yaml.txt](https://github.com/tuhiproject/tuhi/files/3835146/log-1573489775-2019-11-11-17.29.35.yaml.txt)

**Comment by whot** [comment]
from the log:
>   - send: [0xe6, 0x06, 0x96, 0x47, 0xd4, 0x9b, 0x9c, 0xb1]
>   - recv: [0xb3, 0x01, 0x01]

so tuhi sent the connection request and got a "general error" back. which usually means the device wasn't in discovery mode - but on the spark it's expected, see `WacomRegisterHelper.register_device`.

But I don't know why this then disconnected, I wonder if you hit a weird race condition? This is consistent?

**Comment by friedc** [comment]
I have two other connection attempts that went the same way (in fact - these are all the
attempts that are in this directory).

[log-1573480337-2019-11-11-14:52:17.yaml.txt](https://github.com/tuhiproject/tuhi/files/3845564/log-1573480337-2019-11-11-14.52.17.yaml.txt)
[log-1573480380-2019-11-11-14:53:00.yaml.txt](https://github.com/tuhiproject/tuhi/files/3845565/log-1573480380-2019-11-11-14.53.00.yaml.txt)

I also tried to install tuhi directly without flatpak. But the meson that you want is way to new
for bionic (I have meson 0.45.1 - you want meson >= 0.50). The same goes for pygobject
(I have 3.26.1 while you want >= 3.30).

**Comment by whot** [comment]
fwiw, meson is self-contained and you can install it with `pip3 install meson` and then use that. But that obviously doesn't apply to pygobject.

Something is weird here, the two logs - only 1 minute apart, use different uuids. That suggests that we're in the register mode where the uuid is being generated, but then we shouldn't fail after that error, at least not for the spark.

Unfortunately - no idea how to debug this without reproducing it. This will need a lot of printf-style debugging to get to what's going on but that is not something I can do remotely, sorry.

**Comment by niclashoyer** [comment]
I got the same problems and found a workaround (#234). It seems that the `GENERAL_ERROR` that the Spark responds on registration is not correctly catched and transformed into a `AuthorizationError`. Or maybe the `DeviceError` is parsed somewhere else?

**Comment by friedc** [comment]
I am happy to tell you that the problem seems fixed using the PPA build called 0.5-7~bionic from https://launchpad.net/~leo.robol/+archive/ubuntu/tuhi.

Many thanks to @whot, @niclashoyer and @robol !
